click_completion/core.py

```diff
@@ -307,6 +307,111 @@ def get_code(shell=None, prog_name=None, env_name=None, extra_env=None):
+            None (Default value = None)
+        path : str
+            The installation path of the code to be evaluated by the shell. The standard installation path is used if the
+            value is None (Default value = None)
+        append : bool
+            Whether to append the content to the file or to override it. The default behavior depends on the shell type
+            (Default value = None)
+        extra_env : dict
+            A set of environment variables and their values to be added to the generated code (Default value = None)
+
+        Raises
+        -------
+        click.ClickException
+            If the provided Shell isn't supported.
+        """
+        self.prog_name = prog_name or click.get_current_context().find_root().info_name
+        self.shell = shell or get_auto_shell()
+        self.path, self.mode = self.__get_path_config(shell, prog_name, path, append)
+        self.code = get_code(shell, prog_name, env_name, extra_env)
+
+    def __get_path_config(self, shell=None, prog_name=None, path=None, append=None):
+        """
+        Determines the path and write mode for the given shell.
+
+        Parameters
+        ----------
+        shell : Shell
+            The shell type targeted.
+        prog_name : str
+            The program name on the command line.
+        path : str
+            The installation path of the code to be evaluated by the shell.
+        append : bool
+            Whether to append the content to the file or to override it.
+
+        Returns
+        -------
+        Tuple[str, str]
+            The path and write mode for writing the completion code.
+
+        Raises
+        -------
+        click.ClickException
+            If the provided Shell isn't supported.
+
+        """
+        if append is None and path is not None:
+            append = True
+        if append is not None:
+            mode = 'a' if append else 'w'
+        else:
+            mode = None
+
+        if shell == 'fish':
+            path = path or os.path.expanduser('~') + '/.config/fish/completions/%s.fish' % prog_name
+            mode = mode or 'w'
+        elif shell == 'bash':
+            path = path or os.path.expanduser('~') + '/.bash_completion'
+            mode = mode or 'a'
+        elif shell == 'zsh':
+            path = path or os.path.expanduser('~') + '/.zshrc'
+            mode = mode or 'a'
+        elif shell == 'powershell':
+            subprocess.check_call(['powershell', 'Set-ExecutionPolicy Unrestricted -Scope CurrentUser'])
+            path = path or subprocess.check_output(
+                ['powershell', '-NoProfile', 'echo $profile']).strip() if install else ''
+            mode = mode or 'a'
+        else:
+            raise click.ClickException('%s is not supported.' % shell)
+
+        if append is not None:
+            mode = 'a' if append else 'w'
+        else:
+            mode = mode
+
+        return path, mode
```

Comment:
I am fully aware that this overrides the previous configuration setting on the lines above, but since this was the behaviour of the old way I don't want to break it with this PR.